Api/Services/RestaurantService.cs

```diff
@@ -455,5 +455,45 @@ await context
 
             return true;
         }
+        /// <summary>
+        /// Validates if given dto is valid. If a group is given, checks if that group belongs to User
+        /// </summary>
+        /// <param name="dto"></param>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public async Task<Result<bool>> ValidateFirstStepAsync(ValidateRestaurantFirstStepRequest dto, User user)
+        {
+            var errors = new List<ValidationResult>();
+
+            if (!ValidationUtils.TryValidate(dto, errors))
+            {
+                return errors;
+            }
+
+            if (dto.GroupId != null)
+            {
+                var group = await context.RestaurantGroups.FindAsync(dto.GroupId);
+
+                if (group is null)
+                {
+                    errors.Add(new ValidationResult(
+                        $"Group with ID {dto.GroupId} not found",
+                        [nameof(dto.GroupId)]));
+                    return errors;
+                }
+
+                if (group.OwnerId != user.Id)
+                {
+                    errors.Add(new ValidationResult(
+                        $"Group with ID {dto.GroupId} is not owned by the current user",
+                        [nameof(dto.GroupId)]));
+                    return errors;
+                }
```

Comment:
Trzeba jeszcze sprawdzić czy NIP jest ok

Reply:
![image](https://github.com/Reservant-inc/reservant-backend/assets/108064906/98ae7331-4af1-494a-8e58-8a4a537154fe)
![image](https://github.com/Reservant-inc/reservant-backend/assets/108064906/61b898b5-042d-46dd-9f77-705147c1b2cc)

Reply:
A czy to przypadkiem nie sprawdza tylko czy nip jest pusty?

Reply:
sprawdza tak samo jak przy create restaurant

Reply:
ok to nvm